src/once_cell.rs

```diff
@@ -101,3 +101,30 @@ impl<T> GILOnceCell<T> {
         Ok(())
     }
 }
+
+/// Will convert the given value into a Python object and intern that within a [`GILOnceCell`]
+///
+/// # Example
+///
+/// ```
+/// use pyo3::intern;
+/// # use pyo3::{types::PyDict, Python};
+///
+/// # Python::with_gil(|py| {
+///    let dict = PyDict::new(py);
+///    dict.set_item(intern!(py, "foo"), 42).unwrap();
+/// # });
+/// ```
```

Comment:
```suggestion
/// Converts `value` into a Python object and stores it in static storage. The same Python object
/// is returned on each invocation.
///
/// Because it is stored in a static, this object's destructor will not run.
/// 
/// # Example: Using `intern!` to avoid needlessly recreating the same object
///
/// ```
/// use pyo3::intern;
/// # use pyo3::{types::PyDict, Python};
///
/// #[pyfunction]
/// fn create_dict(py: Python<'_>) -> PyResult<&PyDict> {
///    let dict = PyDict::new(py);
///    //             👇 A new `PyString` is created
///    //                for every call of this function
///    dict.set_item("foo", 42)?;
///    Ok(dict)
/// )
///
/// #[pyfunction]
/// fn create_dict_faster(py: Python<'_>) -> PyResult<&PyDict> {
///    let dict = PyDict::new(py);
///    //               👇 A `PyString` is created once and reused
///    //                  for the lifetime of the program.
///    dict.set_item(intern!(py, "foo"), 42)?;
///    Ok(dict)
/// )
/// ```
```

For this one it's important to showcase where you'd want to use it.